compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1152,9 +1152,10 @@ object desugar {
           val tmpName = UniqueName.fresh()
           val patMods =
             mods & Lazy | Synthetic | (if (ctx.owner.isClass) PrivateLocal else EmptyFlags)
+          val zeroSpan = pat.span.withEnd(pat.span.start)
           val firstDef =
             ValDef(tmpName, TypeTree(), matchExpr)
-              .withSpan(pat.span.union(rhs.span)).withMods(patMods)
+              .withSpan(zeroSpan).withMods(patMods)
```

Comment:
Could this be:
```suggestion
              .withSpan(pat.span.startPos).withMods(patMods)
```
This could also possibly be generalized by replacing ValDef by SyntheticValDef and making SyntheticValDef always return a zero-extent span: https://github.com/lampepfl/dotty/blob/8f99dad4664ff28b4d77e0ab0b1e6fe7c590524a/compiler/src/dotty/tools/dotc/ast/tpd.scala#L209-L210

Reply:
Changed to `pat.span.startPos`, the second option seem much more involved :sweat_smile: